code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -96,11 +130,313 @@
 		ticker.mode.update_necro_icons_added(Controller)
 		ticker.mode.update_all_necro_icons()
 		ticker.mode.risen.Add(Controller)
-	name += " ([rand(1,1000)])"
 
-/mob/living/simple_animal/hostile/necro/copy/ListTargets()
-	. = ..()
-	return . - creator
+	if(name == initial(name) && !unique_name)
+		name += " ([rand(1,1000)])"
+
+/mob/living/simple_animal/hostile/necro/zombie/Life()
+	..()
+	/*TODO
+	First, check if the zombie can potentially evolve
+	Have the zombie move to a corpse and start chewing at it
+	If neither of these things are applicable, break some lights to set the mood
+	Otherwise, start wandering and bust down some doors to find more food
+	*/
+	if(!isUnconscious())
```

Comment:
This isn't checking whether it has a user controlling it before doing this automated behavior?